cardinal/server/debug.go

```diff
@@ -25,8 +26,8 @@ func (handler *Handler) registerDebugHandlerSwagger(api *untyped.API) {
 				result := make(DebugStateResponse, 0)
 				search := ecs.NewSearch(filter.All())
 				wCtx := ecs.NewReadOnlyWorldContext(handler.w)
-
-				err := search.Each(wCtx, func(id entity.ID) bool {
+				var err error
+				err = errors.Join(err, search.Each(wCtx, func(id entity.ID) bool {
```

Comment:
redundant join

Reply:
Yeah you're right there was a bug here, but I don't think it was what you were thinking. 

Same thing happening here:
https://github.com/Argus-Labs/world-engine/pull/424#discussion_r1395007644

The bug here is that I'm using `:=` instead of `=` and that is overwriting the err value from outer-scope.